patches/amd-mainline/llvm-project/0001-hipcc-fix-default-include-path-on-Windows-and-adapt-.patch

```diff
@@ -0,0 +1,124 @@
+From 993fefc7d2295fd34219534c5971c50758526d96 Mon Sep 17 00:00:00 2001
+From: Scott Tsai <scottt.tw@gmail.com>
+Date: Tue, 15 Apr 2025 09:30:13 +0800
+Subject: [PATCH 4/4] hipcc: fix default include path on Windows and adapt to
+ TheRock rocm layout
+
+getCppConfig in the orignial code did not add the content of
+`getHipPath()` to the default C Preprocessor definitions in the
+`if (isWindows())` path.
+
+Change default `hipClangPath` to look for clang under `lib/llvm/bin`
+like Linux instead of expecting `clang.exe` to be next to `hipcc.exe`.
+
+Change output of `hipcc.exe --ldflags` to use Windows "\" path
+separators to make the output obviously correct.
+---
+ amd/hipcc/src/hipBin_amd.h | 60 ++++++++++++++------------------------
+ 1 file changed, 22 insertions(+), 38 deletions(-)
+
+diff --git a/amd/hipcc/src/hipBin_amd.h b/amd/hipcc/src/hipBin_amd.h
+index 550e4b248128..f66852b8352d 100644
+--- a/amd/hipcc/src/hipBin_amd.h
++++ b/amd/hipcc/src/hipBin_amd.h
+@@ -202,16 +202,11 @@ void HipBinAmd::constructCompilerPath() {
+   const EnvVariables& envVariables = getEnvVariables();
+   if (envVariables.hipClangPathEnv_.empty()) {
+     fs::path hipClangPath;
+-    if (isWindows()) {
+-      compilerPath = getHipPath();
+-      hipClangPath = compilerPath;
+-      hipClangPath /= "bin";
+-    } else {
+-      compilerPath = getRoccmPath();
+-      hipClangPath = compilerPath;
+-      hipClangPath /= "lib/llvm/bin";
+-    }
```

Comment:
@stellaraccident is aware, but we'll be rolling out more of these compatibility "breaking" changes to better align ROCm across Windows and Linux. If the HIP SDK wants to keep old path structures it could do that in branches.

I think given that, we could upstream this patch to https://github.com/ROCm/llvm-project as well.